app/services/trainees/create_timeline_events.rb

```diff
@@ -91,12 +91,20 @@ def single_event_title
     end
 
     def state_change_title
+      I18n.t("components.timeline.titles.trainee.#{state_change_action}")
+    end
+
+    def state_change_action
       change_from, change_to = audited_changes["state"].map { |change| Trainee.states.key(change) }
 
       if change_from == "deferred" && change_to != "withdrawn"
-        I18n.t("components.timeline.titles.trainee.reinstated")
+        "reinstated"
```

Comment:
I wonder if it'll read a little better if we use a case statement or a hash with the keys matching the expected `change_to` targets with their corresponding values?

Reply:
I fiddled around with this for a while - but the first is mostly based on the `change_from` value, not the `change_to`.

I would have to guard against the first conditional, and then have a case statement i think.